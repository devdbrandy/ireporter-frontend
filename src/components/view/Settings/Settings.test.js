import React from 'react';
import { shallow } from 'enzyme';
import { Settings } from './index';

/**
 * Wrapper for enzyme shallow component
 *
 * @returns {object} The mocked props and component
 */
function setup() {
  const props = {
    user: {},
  };

  const component = shallow(<Settings {...props} />);
  return { props, component };
}

describe('<Settings />', () => {
  it('renders without crashing given the required props', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
