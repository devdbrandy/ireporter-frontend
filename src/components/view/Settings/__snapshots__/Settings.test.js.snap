// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> renders without crashing given the required props 1`] = `
<Fragment>
  <Layout>
    <div
      className="container"
    >
      <main
        id="main"
      >
        <div
          className="column"
        >
          <div
            className="panel"
          >
            <div
              className="panel--heading"
            >
              <h3>
                Settings
              </h3>
            </div>
            <div
              className="panel--body"
            >
              <form
                id="settings-form"
              >
                <div
                  className="form-control"
                >
                  <div
                    className="col-2"
                  >
                    <label
                      htmlFor="username"
                    >
                      Username
                    </label>
                  </div>
                  <div
                    className="col-10"
                  >
                    <input
                      disabled={true}
                      id="username"
                      type="text"
                      value="@username"
                    />
                  </div>
                </div>
                <div
                  className="form-control"
                >
                  <div
                    className="col-2"
                  >
                    <label
                      htmlFor="email"
                    >
                      Email Address
                    </label>
                  </div>
                  <div
                    className="col-10"
                  >
                    <input
                      disabled={true}
                      id="email"
                      type="email"
                      value="johndoe@email.com"
                    />
                  </div>
                </div>
                <div
                  className="form-control"
                >
                  <div
                    className="col-2"
                  >
                    <label
                      htmlFor="firstname"
                    >
                      Name
                    </label>
                  </div>
                  <div
                    className="col-10"
                  >
                    <div
                      className="row"
                    >
                      <div
                        className="column pr-4"
                      >
                        <input
                          id="firstname"
                          name="firstname"
                          type="text"
                          value="John"
                        />
                      </div>
                      <div
                        className="column pl-4 pr-4"
                      >
                        <input
                          id="lastname"
                          name="lastname"
                          type="text"
                          value="Doe"
                        />
                      </div>
                    </div>
                    <div
                      className="row"
                    >
                      <input
                        id="othernames"
                        name="othernames"
                        placeholder="Other names"
                        type="text"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="form-control"
                >
                  <div
                    className="col-2"
                  >
                    <label
                      htmlFor="phonenumber"
                    >
                      Phone number
                    </label>
                  </div>
                  <div
                    className="col-10"
                  >
                    <input
                      id="phonenumber"
                      name="phoneNumber"
                      type="text"
                      value="07012345678"
                    />
                  </div>
                </div>
                <div
                  className="form-control"
                >
                  <div
                    className="col-2"
                  >
                    <label
                      htmlFor="bio"
                    >
                      Summary (Bio)
                    </label>
                  </div>
                  <div
                    className="col-10"
                  >
                    <textarea
                      id="bio"
                      name="bio"
                      placeholder="a short bio of yourself"
                    >
                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quibusdam assumenda doloremque ab! Enim debitis soluta dignissimos illum veritatis cumque non.
                    </textarea>
                  </div>
                </div>
                <div
                  className="form-control"
                >
                  <div
                    className="col-2"
                  >
                    <label
                      htmlFor="title"
                    >
                      Profile Image
                    </label>
                  </div>
                  <div
                    className="col-10"
                  >
                    <div
                      className="row middle"
                    >
                      <div
                        className="profile-image"
                      >
                        <img
                          alt="avatar"
                          src="img/avatar.jpg"
                        />
                      </div>
                      <input
                        id="avatar"
                        name="avatar"
                        type="hidden"
                      />
                      <button
                        className="btn btn-primary btn-sm ml-20"
                        id="upload-avatar"
                        type="button"
                      >
                        Upload avatar
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  className="form-control"
                >
                  <div
                    className="col-2"
                  >
                    <label
                      htmlFor="gender"
                    >
                      Gender
                    </label>
                  </div>
                  <div
                    className="col-10"
                  >
                    <label>
                      <input
                        name="gender"
                        type="radio"
                        value="male"
                      />
                      Male
                    </label>
                    <label>
                      <input
                        name="gender"
                        type="radio"
                        value="female"
                      />
                      Female
                    </label>
                  </div>
                </div>
                <div
                  className="row center mt-60"
                >
                  <button
                    className="btn btn-primary"
                    type="submit"
                  >
                    Save Settings
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </main>
    </div>
  </Layout>
</Fragment>
`;
